test/e2e/util_gatewayapi_test.go

```diff
@@ -562,27 +563,42 @@ func assertGatewaySuccessful(t *testing.T, namespace, name string) (*gatewayapiv
 
 	gw := &gatewayapiv1.Gateway{}
 	nsName := types.NamespacedName{Namespace: namespace, Name: name}
-	recordedConditionMsg := "not found"
+	recordedAcceptedConditionMsg, recordedProgrammedConditionMsg := "not found", "not found"
```

Comment:
With your change below to use `%q` instead of `%s`, it would be clearer to initialize these variables to empty strings.